src/main/java/edu/cornell/kfs/ksr/service/impl/SecurityRequestDerivedRoleTypeServiceImpl.java

```diff
@@ -230,9 +228,14 @@ protected boolean isQualifiedAuthorizerRole(String authorizerRoleId) {
 
         return false;
     }
-
-    protected BusinessObjectService geBusinessObjectService() {
-        return KRADServiceLocator.getBusinessObjectService();
+    
+    @Override
+    public boolean dynamicRoleMembership(String namespaceCode, String roleName) {

```

Comment:
I added this override per your notes on the parent user story.  If you think it would be preferable to add the checks in I can.  Since the goal of this over ride is to prevent caching, I don't think we want those checks, but I would go with what you think is the preferable approach.  Please let me know

Reply:
I think it would be good to have the blank-value checks in place, for consistency with base code.